code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -104,7 +100,8 @@ func (zcn *ZCNSmartContract) AddAuthorizer(tran *transaction.Transaction, inputD
 
 	// Dig pool for authorizer
 
-	transfer, response, err := authorizer.Staking.DigPool(tran.Hash, tran)
+	var transfer *state.Transfer
```

Comment:
If I understand what is happening here correctly all authorizers should start with a default delegate. 

In which case this locking pool needs to be replaced with a stakepoolStakePool object. Then instead of calling `LockingPool.DigPoo`l, we can call [StakePool.Lock](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/stakepool/lock.go#L44). 

The values for [StakePool.Settings](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/stakepool/stakepool.go#L29) can either be added to the input or inherited from the Global node. The global node settings are the bonds on individual authoriser stakePool settings.

This way an authoriser always has a stakePool hence `updateAuthorizerStakePool` not `AddOrUpdateAuthorizerStakePool`.

Reply:
Done